dbgap_extract.py

```diff
@@ -161,10 +202,50 @@ def get_sra_data_details_from_xml_sample(sample):
     return sra_data_details
 
 
-def get_sample_dict_from_xml_sample(study_accession, sample):
+def _get_sra_data_details_from_xml_sample(sample):
+    """
+    Get SRA details as a dictionary
+
+    Args:
+        sample (xml.etree.ElementTree.Element): sample element from XML tree
+
+    Returns:
+        dict: SRA details as a dict
+    """
+    sra_datas = sample.findall("SRAData")[0].findall("Stats")
+    sra_data_details = {}
+    if len(sra_datas) > 0:
+        for stat in sra_datas:
+            stat_dict = stat.attrib
+            sra_data_details.update(stat_dict)
+    sra_details = json.dumps(sra_data_details).replace('""', '"')
+    return sra_details
+
+
+def get_sample_dict_from_xml_sample(study_accession, sample, args):
+    """
+    Get dictionary of sample from the XML sample
+
+    Args:
+        study_accession (str): the study phsid/accession number
+        sample (xml.etree.ElementTree.Element): sample element from XML tree
+        args (argparse.Namespace): arguments sent to command line
+
+    Returns:
+        dict: SRA details as a dict
+    """
     sample_dict = sample.attrib
-    sample_dict["sample_use"] = get_sample_use_from_xml_sample(sample)
-    sample_dict["sra_data_details"] = get_sra_data_details_from_xml_sample(sample)
+
+    sample_dict["sample_use"] = json.dumps(
+        [use.text for use in sample.findall("Uses")[0].findall("Use")]
```

Comment:
this one-liner leaves an empty list "[]" in the fields rather than an empty cell -- is that intentional? it differs from what master branch does

Reply:
it is intentional, I made the sra details optionally a column filled with valid json and the same for this column. so tools can safely json.load the contents